pkg/ingress/controller/controller.go

```diff
@@ -728,8 +740,13 @@ func (c *Controller) ensureIngress(ing *nwv1.Ingress) error {
 
 	// Create listener
 	sourceRanges := getStringFromIngressAnnotation(ing, IngressAnnotationSourceRangesKey, "0.0.0.0/0")
+	timeoutClientData := getIntFromIngressAnnotation(ing, IngressAnnotationTimeoutClientData, 50000)
+	timeoutMemberConnect := getIntFromIngressAnnotation(ing, IngressAnnotationTimeoutMemberConnect, 5000)
+	timeoutMemberData := getIntFromIngressAnnotation(ing, IngressAnnotationTimeoutMemberData, 50000)
+	timeoutTCPInspect := getIntFromIngressAnnotation(ing, IngressAnnotationTimeoutTCPInspect, 0)
```

Comment:
I think that one requirement for this change should be: **if the new annotations are not set in config, the Ingress behaves exactly as it was before this patch**.

I am worried that setting an explicit default timeout changes the default behaviour. If no annotation is set, `TimeoutXxx` fields in `listeners.CreateOpts` should be left nil as they were previously.

Reply:
Understandable, thanks @pierreprinetti I have made the requested changes